connext_cmake_module/cmake/Modules/FindConnext.cmake

```diff
@@ -211,14 +211,10 @@ if(NOT "${_NDDSHOME} " STREQUAL " ")
     set(Connext_DDSGEN2 "rtiddsgen2.bat")
   else()
     set(Connext_DEFINITIONS "RTI_LINUX" "RTI_UNIX")
-    if(EXISTS "${Connext_HOME}/bin/rtiddsgen2")
-      set(Connext_DDSGEN2 "${Connext_HOME}/bin/rtiddsgen2")
-    elseif(EXISTS "${Connext_HOME}/scripts/rtiddsgen2")
-      set(Connext_DDSGEN2 "${Connext_HOME}/scripts/rtiddsgen2")
-    elseif(EXISTS "${Connext_HOME}/bin/rtiddsgen")
-      set(Connext_DDSGEN2 "${Connext_HOME}/bin/rtiddsgen")
+    if(EXISTS "${Connext_HOME}/bin/rtiddsgen_server")
+      set(Connext_DDSGEN2 "${Connext_HOME}/bin/rtiddsgen_server")
```

Comment:
I have updated the `connext_cmake_module` to always export the variable pointing to `rtigendds` and if it exists a export a second variable pointing to the server mode equivalent.

The downstream package `rosidl_typesupport_connext_cpp` then chooses whats available with a comment why it favors the server one.